desktop/libs/notebook/src/notebook/connectors/hiveserver2.py

```diff
@@ -287,6 +287,8 @@ def close_session_idle(self, notebook, session):
 
   @query_error_handler
   def execute(self, notebook, snippet):
+    if snippet['type'] == 'sql':
```

Comment:
Thanks for the PR!
 
Could we just move it down in the `def _get_db(self, snippet, is_async=False, interpreter=None):` function? (more contained/cleaner there)

Reply:
Actually I found override snippet['type'] will cause new issue when right click table and select "Open in Editor" in "tables" tab
```
[24/Oct/2020 10:05:44 -0700] middleware   INFO     Processing exception: Snippet type sparksql is not configured.: Traceback (most recent call last):
  File "/usr/lib/hue/build/env/lib/python2.7/site-packages/Django-1.11.29-py2.7.egg/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib/hue/build/env/lib/python2.7/site-packages/Django-1.11.29-py2.7.egg/django/utils/decorators.py", line 185, in inner
    return func(*args, **kwargs)
  File "/usr/lib/hue/desktop/libs/notebook/src/notebook/views.py", line 186, in browse
    return JsonResponse(notebook.execute(request, batch=False))
  File "/usr/lib/hue/desktop/libs/notebook/src/notebook/connectors/base.py", line 291, in execute
    return _execute_notebook(request, notebook_data, snippet)
  File "/usr/lib/hue/desktop/libs/notebook/src/notebook/api.py", line 145, in _execute_notebook
    interpreter = get_api(request, snippet)
  File "/usr/lib/hue/desktop/libs/notebook/src/notebook/models.py", line 516, in get_api
    return ApiWrapper(request, snippet)
  File "/usr/lib/hue/desktop/libs/notebook/src/notebook/models.py", line 502, in __init__
    self.api = _get_api(request, snippet)
  File "/usr/lib/hue/desktop/libs/notebook/src/notebook/connectors/base.py", line 415, in get_api
    interpreter = get_interpreter(connector_type=connector_name, user=request.user)
  File "/usr/lib/hue/desktop/libs/notebook/src/notebook/connectors/base.py", line 380, in get_interpreter
    raise PopupException(_('Snippet type %s is not configured.') % connector_type)
PopupException: Snippet type sparksql is not configured.
```

I think we should do `application='sql'` when `self.query_server['server_name'] == 'sparksql'` to minimize the impact since the root cause is when creating session objects, we are using server name `sparksql`
https://github.com/cloudera/hue/blob/524c489d07ca410c6ac11fbbe060b5ff80f65f94/apps/beeswax/src/beeswax/server/hive_server2_lib.py#L695